packages/plugin-color/src/Color.ts

```diff
@@ -0,0 +1,166 @@
+import { JWPlugin, JWPluginConfig } from "../../core/src/JWPlugin";
+import { CommandParams } from "../../core/src/Dispatcher";
+import { Inline } from "../../plugin-inline/src/Inline";
+import { VNode } from "../../core/src/VNodes/VNode";
+import { getStyle, setStyle, removeStyle } from "../../utils/src/utils";
+import { InlineNode } from "../../plugin-inline/src/InlineNode";
+import { FragmentNode } from "../../core/src/VNodes/FragmentNode";
+import { VRange } from "../../core/src/VRange";
+import { Format } from "../../plugin-inline/src/Format";
+
+export interface ColorParams extends CommandParams {
+    color?: string; // css-valid color name
+}
+export interface ColorConfig extends JWPluginConfig {
+    defaultColor?: string; // css-valid color name
+}
+
+export class Color<T extends ColorConfig = ColorConfig> extends JWPlugin<T> {
+    static dependencies = [Inline];
+    styleName: string; // css-valid style property name (eg: 'background-color')
+
+    //--------------------------------------------------------------------------
+    // Public
+    //--------------------------------------------------------------------------
+
+    /**
+     * Return true if the given node has the given color.
+     * If no color is passed, return true if the given node has any color.
+     * Return false otherwise.
+     */
+    hasColor(node: VNode): boolean;
+    hasColor(color: string, node: VNode): boolean;
+    hasColor(color: string | VNode, node?: VNode): boolean {
```

Comment:
I think you'll be srewed when you try to create the color buttons because this function is not static anymore.

Reply:
Yes I figured as much. But actually this is not the function I should use for the buttons : I think the buttons should check the color of the first ancestor/format that has a color. So we'll implement that when we do have buttons...